src/DotLiquid/StandardFilters.cs

```diff
@@ -805,10 +846,23 @@ public static object Ceil(Context context, object input)
         /// <returns>The rounded value; null if an exception have occurred</returns>
         public static object Floor(Context context, object input)
         {
-            if (decimal.TryParse(input.ToString(), NumberStyles.Any, context.CurrentCulture, out decimal d))
-                return Math.Floor(d);
-            else
-                return null;
+            if (context.SyntaxCompatibilityLevel < SyntaxCompatibility.DotLiquid22b)

```

Comment:
I wonder if we should make a new attribute on the method specifying the Liquid compatibility, such as:

```c#
public static object Floor(Context context, object input)
        {
        ...
        }

[LiquidCompatibility(name:="Floor",minVersion=SyntaxCompatibility.DotLiquid24]
public static object FloorV24(Context context, object input)
        {
        ...
        }
```
This would then be utilized by Strainer.Create to ignore the incompatible ones. Caveat: Param tag would be unable to change the compatibility in middle of template.

Reply:
Give me a day or two to mock up how this would be. I'm wary to add more conditionals and want to test the performance impact.

Reply:
Oooh, very interesting idea. I like that we wouldn't have a bunch of if conditions in the middle of a method implementing a Liquid filter. Much better code readability, and easier to avoid mistakes regressing earlier compatibility implementations.

Reply:
See #577 @daviburg , it does cleanup DotLiquid base, but I think it makes DotLiquid.Tests messier as direct method calls do not trigger this logic.

Reply:
I like this idea too. Makes it much more self-documenting. 

Tests could do something similar to Strainer to return the version of a function matching a compatibility level, and you could annotate the tests in a similar way, and match them up! You'd need separate test functions for the specific versions of functions where things change, but that will also make it more obvious when syntax has changed. I find reading tests is a great way to understand how to use a code base. 

I don't suppose this extends to non-filter changes in syntax (e.g. there are checks in Context and Tokenizer). But it's an improvement.